EEG101/src/components/NavBar.js

```diff
@@ -31,8 +39,151 @@ function mapDispatchToProps(dispatch) {
 class NavBar extends Component {
   constructor(props) {
     super(props);
+    this.predictSubscription = {};
+	
+    this.state = {
+      batteryLevel: '',
+    };
+  }
+  
+  componentDidMount () {
+    TimerMixin.setInterval(() => { 
+	  this.props.setRefresh(!this.props.refresh); 
+	  this.startBatteryReading();
+      this.stopBatteryReading();
+	}, 1000);
+  }
+  
+   componentWillUnmount() {
+    this.props.setNoMore(this.props.noMore = false);
+  }
+
+  startBatteryReading() {
+    if (this.props.connectionStatus === config.connectionStatus.CONNECTED) {
+      if (this.props.noMore === false) {
+        Battery.startReading();
+        const batteryListener = new NativeEventEmitter(
+          NativeModules.Battery
+        );
+        this.predictSubscription = batteryListener.addListener(
+          "BATTERY", battery => {
+            this.state.batteryLevel = battery;
+          }
+        );
+        this.props.setNoMore(this.props.noMore = true);
+      } else {
+        return null;
+      }
+    } else {
+        return null;
+    }
+  }
+
+  stopBatteryReading() {
+    if (this.props.connectionStatus === config.connectionStatus.DISCONNECTED) {
+      if (this.props.noMore === true) {
+        Battery.stopReading();
+        const batteryListener = new NativeEventEmitter(
+          NativeModules.Battery
+        );
+        this.predictSubscription = batteryListener.removeListener();
+        this.props.setNoMore(this.props.noMore = false);
+      } else {
+        return null;
+      }
+    } else {
+        return null;
+    }
+  }
+
+  renderTrick() {
+    if (this.props.refresh === true) {
+      return (
+        <View>
+        </View>
+      );
+    } else {
+      return (
+        <View>
+        </View>
+      );
+    }
   }
 
+  renderBatteryIcon() {
+    if (this.props.connectionStatus === config.connectionStatus.CONNECTED) {
+      if (this.state.batteryLevel === '') {
```

Comment:
I think it'd be more readable to have this written as a switch statement

Reply:
As I mentioned I'm not professional programmer, so I wrote it as I could ;)